packages/lesswrong/components/ultraFeed/UltraFeedCommentItem.tsx

```diff
@@ -0,0 +1,290 @@
+      maxLevelReached: maxReached,
+      wordCount,
+    });
+
+    // If the comment was previously collapsed and is now expanding (level > 0),
+    // update its display status in the parent component.
+    if (displayStatus === "collapsed" && level > 0) {
+      onChangeDisplayStatus("expanded");
+    }
+
+  }, [trackExpansion, comment._id, comment.postId, displayStatus, onChangeDisplayStatus]);
+
+  const expanded = displayStatus === "expanded";
+
+  const truncationBreakpoints = useMemo(() => {
+    const fullBreakpoints = settings.commentTruncationBreakpoints || [];
+    const collapsedLimit = settings.collapsedCommentTruncation;
+
+  if (!post) {
+    // eslint-disable-next-line no-console
+    console.log("Missing post data:", comment._id);
+    return null;
+  }
+
+    if (expanded) {
+      return fullBreakpoints;
+    } else {
+      // If collapsed, use the collapsed limit as the first breakpoint,
+      // followed by the rest of the standard breakpoints (skipping the first standard one).
+      const subsequentBreakpoints = fullBreakpoints.length > 1 ? fullBreakpoints.slice(1) : [];
+      return [collapsedLimit, ...subsequentBreakpoints];
+    }
+  }, [expanded, settings.commentTruncationBreakpoints, settings.collapsedCommentTruncation, post, comment._id]);
+
+  const shouldUseLineClamp = !expanded && settings.lineClampNumberOfLines > 0;
+
+  return (
+    <div ref={elementRef} className={classNames(classes.root)} >
+      <div className={classes.verticalLineContainer}>
+        <div className={classNames(
+          classes.verticalLine,
+          { 
+            [classes.verticalLineHighlighted]: highlight,
+            [classes.verticalLineFirstComment]: isFirstComment,
+            [classes.verticalLineLastComment]: isLastComment
+          }
+        )} />
+      </div>
+      
+      <div className={classNames(classes.commentContentWrapper, { [classes.commentContentWrapperWithBorder]: !isLastComment })}>
+        <div className={classes.commentHeader}>
+          <UltraFeedCommentsItemMeta comment={comment} />
+          {showInLineCommentThreadTitle && !comment.shortform && post && (
+            <div className={classes.inlineCommentThreadTitle}>
+              <button 
+                className={classes.inlineCommentThreadTitleLink}
+                onClick={onPostTitleClick}
+              >
+                Replying to <span className={classes.inlineCommentThreadTitleLinkSpan}>{post.title}</span>
+              </button>
+            </div>
+          )}
+        </div>
+        <div className={classes.contentWrapper}>
+          <FeedContentBody
+            comment={comment}
+            html={comment.contents?.html || ""}
+            breakpoints={truncationBreakpoints || []}
+            wordCount={comment.contents?.wordCount || 0}
+            linkToDocumentOnFinalExpand={expanded}
+            initialExpansionLevel={0}
+            nofollow={(comment.user?.karma || 0) < nofollowKarmaThreshold.get()}
```

Comment:
I think all of these should be `??`.  (In none of these cases does it matter, but on the rare occasions where it does matter, `??` is almost always the one you want, unless you're explicitly trying to avoid ending up with 0 or empty string and the right side is some value other than those.)